test/powershell/Modules/Microsoft.PowerShell.Management/Set-Service.Tests.ps1

```diff
@@ -0,0 +1,172 @@
+Describe "Set/New-Service cmdlet tests" -Tags "Feature", "RequireAdminOnWindows" {
+    BeforeAll {
+        $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+        if ( -not $IsWindows ) {
+            $PSDefaultParameterValues["it:skip"] = $true
+        }
+    }
+    AfterAll {
+        $global:PSDefaultParameterValues = $originalDefaultParameterValues
+    }
+
+    It "SetServiceCommand can be used as API for '<parameter>' with '<value>'" -TestCases @(
+        @{parameter = "ComputerName"; value = "foo"},
+        @{parameter = "Name"        ; value = "bar"},
+        @{parameter = "DisplayName" ; value = "hello"},
+        @{parameter = "Description" ; value = "hello world"},
+        @{parameter = "StartupType" ; value = "Automatic"},
+        @{parameter = "StartupType" ; value = "Boot"},
+        @{parameter = "StartupType" ; value = "Disabled"},
+        @{parameter = "StartupType" ; value = "Manual"},
+        @{parameter = "StartupType" ; value = "System"},
+        @{parameter = "Status"      ; value = "Running"},
+        @{parameter = "Status"      ; value = "Stopped"},
+        @{parameter = "Status"      ; value = "Paused"},
+        @{parameter = "InputObject" ; value = (Get-Service | Select-Object -First 1)},
+        # cmdlet inherits this property, but it's not exposed as parameter so it should be $null
+        @{parameter = "Include"     ; value = "foo", "bar" ; expectedNull = $true},
+        # cmdlet inherits this property, but it's not exposed as parameter so it should be $null
+        @{parameter = "Exclude"     ; value = "foo", "bar" ; expectedNull = $true}
+    ) {
+        param($parameter, $value, $expectedNull)
+
+        $setServiceCommand = [Microsoft.PowerShell.Commands.SetServiceCommand]::new()
+        $setServiceCommand.$parameter = $value
+        if ($expectedNull -eq $true) {
+            $setServiceCommand.$parameter | Should BeNullOrEmpty
+        }
+        else {
+            $setServiceCommand.$parameter | Should Be $value
+        }
+    }
+
+    It "Set-Service parameter validation for invalid values: <script>" -TestCases @(
+        @{script    = {Set-Service foo -StartupType bar};
+            errorid = "CannotConvertArgumentNoMessage,Microsoft.PowerShell.Commands.SetServiceCommand"
```

Comment:
Formatting.

Reply:
will fix, I see another issue lower that I'll fix as well